posthog/hogql/escape_sql.py

```diff
@@ -44,6 +46,8 @@ def escape_hogql_identifier(identifier: str | int) -> str:
 
 # Copied from clickhouse_driver.util.escape, adapted from single quotes to backquotes.
 def escape_clickhouse_identifier(identifier: str) -> str:
+    if "%" in identifier:
+        raise HogQLException(f'The HogQL identifier "{identifier}" is not permitted as it contains the "%" character')
```

Comment:
in this case, should the error name be Clickhouse identifier to stay consistent with dialect being used?

Reply:
Not really, since we're showing this error directly to the users, and don't want to bother them with such nuance :).